benchmark/src/benchmark.ts

```diff
@@ -0,0 +1,82 @@
+import { isEqual } from 'lodash-es';
+
+export interface BenchmarkValue<T> {
+    /**
+     * The value returned by the benchmarked function
+     */
+    value: T;
+    /**
+     * The average time it took to run the function in ms
+     */
+    time: number;
+}
+
+/**
+ * @param functionToBenchmark the deterministic function that should be benchmarked, it will be run multiple times
+ * @param options additional options for the benchmark, the defaults are {@link defaultOptions}
+ */
+export function benchmark<Value>(
+    functionToBenchmark: () => Value,
+    options: BenchmarkOptions<Value> = {}
+): BenchmarkValue<Value> {
+    // eslint-disable-next-line no-param-reassign
+    options = { ...defaultOptions, ...options };
+    const benchmarkStepValues: BenchmarkValue<Value>[] = [];
+    for (let i = 0; i < options.numberOfIterations!; i++) {
+        const benchmarkStepValue = runBenchmarkOnce(functionToBenchmark);
+        benchmarkStepValues.push(benchmarkStepValue);
+        options.onAfterIteration?.(benchmarkStepValue);
+    }
+    if (
+        benchmarkStepValues.length > 1 &&
+        benchmarkStepValues.some(
+            ({ value }) => !isEqual(benchmarkStepValues[0]!.value, value)
+        )
+    ) {
+        options.onNonDeterministicError?.(benchmarkStepValues);
+    }
+    const endResult: BenchmarkValue<Value> = {
```

Comment:
```suggestion
    const endResult = {
```

I don't think this is needed.